sockeye/inference.py

```diff
@@ -511,35 +535,151 @@ def get_max_output_length(input_length: int):
+        """
+        return 1 + (0 if not self.factors else len(self.factors))
+
+    def chunks(self, chunk_size: int) -> Generator['TranslatorInput', None, None]:
+        """
+        Takes a TranslatorInput (itself) and yields TranslatorInputs for chunks of size chunk_size.
+
+        :param chunk_size: The maximum size of a chunk.
+        :return: A generator of TranslatorInputs, one for each chunk created.
+        """
+        for chunk_id, i in enumerate(range(0, len(self), chunk_size)):
+            factors = [factor[i:i + chunk_size] for factor in self.factors] if self.factors is not None else None
+            yield TranslatorInput(sentence_id=self.sentence_id,
+                                  tokens=self.tokens[i:i + chunk_size],
+                                  factors=factors,
+                                  chunk_id=chunk_id)
+
+
+def make_input_from_plain_string(sentence_id: int, string: str) -> TranslatorInput:
+    """
+    Returns a TranslatorInput object from a plain string.
+
+    :param sentence_id: An integer id.
+    :param string: An input string.
+    :return: A TranslatorInput.
+    """
+    return TranslatorInput(sentence_id, tokens=list(data_io.get_tokens(string)), factors=None)
+
+
+def make_input_from_json_string(sentence_id: int, json_string: str) -> TranslatorInput:
+    """
+    Returns a TranslatorInput object from a JSON object, serialized as a string.
+
+    :param sentence_id: An integer id.
+    :param json_string: A JSON object serialized as a string.
+    :return: A TranslatorInput.
+    """
+    jobj = json.loads(json_string, encoding=C.JSON_ENCODING)
+    tokens = jobj.get(C.JSON_TEXT_KEY)
+    if tokens is None:
+        raise ValueError("Missing 'text' field in JSON string: %s", json_string)
+
+    tokens = list(data_io.get_tokens(tokens))
+    factors = jobj.get(C.JSON_FACTORS_KEY)
+    if isinstance(factors, list):
+        factors = [list(data_io.get_tokens(factor)) for factor in factors]
+    else:
+        factors = None
+
+    return TranslatorInput(sentence_id=sentence_id, tokens=tokens, factors=factors)
+
+
+def make_input_from_factored_string(sentence_id: int,
+                                    factored_string: str,
+                                    translator: 'Translator',
+                                    delimiter: str = C.DEFAULT_FACTOR_DELIMITER) -> TranslatorInput:
+    """
+    Returns a TranslatorInput object from a string with factor annotations on a token level, separated by delimiter.
+    If translator does not require any source factors, the string is parsed as a plain token string.
+
+    :param sentence_id: An integer id.
+    :param factored_string: An input string with additional factors per token, separated by delimiter.
+    :param translator: A translator object.
+    :param delimiter: A factor delimiter. Default: '|'.
+    :return: A TranslatorInput.
+    """
+    utils.check_condition(bool(delimiter) and not delimiter.isspace(),
```

Comment:
Do we need explicit casting to bool? Doesn't python do it automatically (and AFAIK is recommended)?

Reply:
its for mypi... :)